proto/regen/group/v1alpha1/types.proto

```diff
@@ -0,0 +1,266 @@
+syntax = "proto3";
+
+package regen.group.v1alpha1;
+
+option go_package = "github.com/regen-network/regen-ledger/x/group";
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+import "cosmos_proto/cosmos.proto";
+import "google/protobuf/any.proto";
+
+// Member represents a group member with an account address,
+// non-zero power and a comment.
+message Member {
+
+    // address is the member's account address.
+    string address = 1;
+    
+    // power is the member's power that should be greater than 0.
+    string power = 2;
+    
+    // comment is the member's comment.
+    string comment = 3;
+}
+
+// ThresholdDecisionPolicy implements the DecisionPolicy interface
+message ThresholdDecisionPolicy {
+    option (cosmos_proto.implements_interface) = "DecisionPolicy";
+
+    // threshold is a fix weight value that must be met or exceeded for a proposal to succeed.
+    string threshold = 1;
+    
+    // timeout is the duration from submission of a proposal to the end of voting period
+    // Within this times votes and exec messages can be submitted.
+    google.protobuf.Duration timeout = 2 [(gogoproto.nullable) = false];
+}
+
+// Choice defines available types of choices for voting.
+enum Choice {
+
+    // CHOICE_UNSPECIFIED defines a no-op voting choice.
+    CHOICE_UNSPECIFIED = 0;
+
+    // CHOICE_NO defines a no voting choice.
+    CHOICE_NO = 1;
+
+    // CHOICE_YES defines a yes voting choice.
+    CHOICE_YES = 2;
+
+    // CHOICE_ABSTAIN defines an abstaining voting choice.
+    CHOICE_ABSTAIN = 3;
+
+    // CHOICE_VETO defines a voting choice with veto.
+    CHOICE_VETO = 4;
+}
+
+//
+// State
+//
+
+// GroupInfo represents the high-level on-chain information for a group.
+message GroupInfo {
+
+    // group_id is the unique ID of the group.
+    uint64 group_id = 1 [(gogoproto.casttype) = "ID"];
+
+    // admin is the account address of the group's admin.
+    string admin = 2;
+    
+    // comment is the group's comment.
+    string comment = 3;
+
+    // version is used to track changes to a group's membership structure that
+    // would break existing proposals. Whenever any members power is changed,
+    // or any member is added or removed this version is incremented and will
+    // cause proposals based on older versions of this group to fail
+    uint64 version = 4;
+
+    // total_weight is the sum of the group members' powers.
+    string total_weight = 5;
+}
+
+// GroupMember represents the relationship between a group and a member.
+message GroupMember {
```

Comment:
I believe the type `GroupMember` is only used in query responses and should just be in that file, maybe called `GroupMemberPair` or something... That would address these TODOs right? And we can just embed `Member` inside here to reduce types.

Reply:
it's also used for the `groupMemberTable`, I think `GroupMember` is enough to show that it represents the actual relationship between a group and a member. But I guess we can embed `Member` inside it instead of repeating individual `Member`'s fields.